src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -663,49 +687,57 @@ char* java_lang_String::as_utf8_string_full(oop java_string, char* buf, int bufl
   }
 }
 
-char* java_lang_String::as_utf8_string(oop java_string, typeArrayOop value, char* buf, int buflen) {
+char* java_lang_String::as_utf8_string(oop java_string, typeArrayOop value, char* buf, size_t buflen) {
   assert(value_equals(value, java_lang_String::value(java_string)),
          "value must be same as java_lang_String::value(java_string)");
-  int     length = java_lang_String::length(java_string, value);
+  // `length` is used as the incoming number of characters to
+  // convert, and then set as the number of bytes in the UTF8 sequence.
+  size_t  length = java_lang_String::length(java_string, value);
```

Comment:
Above comment looks wrong.  `length` is not an in/out reference below, so why not leave it as `int` at line 695?  Assigning `int` to `size_t` will introduce a -Wsign-conversion warning.

Reply:
Thanks for looking at this @dean-long ! Good catch. Two places in the `String` code had leftovers from an earlier incarnation of the fix.